reports/aksVersions/aks-versions.sh

```diff
@@ -0,0 +1,147 @@
+  subscriptions+=($sub)
+done
+
+echo "Subscriptions found:"
+for sub in "${subscriptions[@]}"
+do
+  echo "$sub"
+done
+
+# # ---------------------------------------------------------------------------
+# # STEP 2:
+# # ---------------------------------------------------------------------------
+# # Use subscription list to search for AKS Clusters and add to aksClusterInfo array
+# # Use the aksClusterInfo array to find available upgrades and build a new object for each cluster containing the version information and status 
+# # --------------------------------------------------------------------------
+
+declare -a aksClusterInfo
+
+for sub in "${subscriptions[@]}"; do
+  
+  declare -a currentClusters
+
+  echo "Setting subscription: $sub"
+  az account set --subscription "$sub"
+  echo "Searching for AKS clusters..."
+  currentClusters+=$(az aks list --query '[*].{"Name":name,"ResourceGroup":resourceGroup, "kubernetesVersion":kubernetesVersion, "PowerState":powerState.code}' | jq .)
+  echo "Done searching!"
+
+  for cluster in $(echo "$currentClusters" | jq -c '.[]'); do
+    resourceGroup=$(echo "$cluster" | jq -r '.ResourceGroup')   
+    aksName=$(echo "$cluster" | jq -r '.Name')
+
+    echo "Checking for upgrades to: $(echo $cluster | jq -r '.Name')"
+    currentVersion=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.kubernetesVersion')
+    upgradeAvailable=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.upgrades')
+
+    if [[ $upgradeAvailable == 'null' ]]; then
+      verdict="No Update Required"
+      color_code=green
+      upgradeableVersion="None"
+    else
+      upgradesIsPreview=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.upgrades[].isPreview')
+      if [[ $upgradesIsPreview == 'null' ]]; then
+        upgradeableVersion=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.upgrades[].kubernetesVersion')
+        color_code=red
+        verdict="Update required"
+      else
+        upgradeableVersion="Preview Only"
+        color_code=orange
+        verdict="Not ready for update"
+      fi
+    fi
+
+    echo "Creating json object for: $(echo $cluster | jq -r '.Name')"
+    aksClusterInfo+=$(echo "$cluster" | jq -r '.' | jq --arg verdict "$verdict" \
+                                                      --arg color_code "$color_code" \
+                                                      --arg currentVersion "$currentVersion" \
+                                                      --arg id "$(uuidgen)" \
+                                                      --arg upgradeableVersion "$upgradeableVersion" '{id: $id, clusterName: .Name, powerState: .PowerState, currentVersion: $currentVersion, upgradeableVersion: $upgradeableVersion, colorCode: $color_code,  verdict: $verdict, resourceType: "AKS Cluster"}')  
+  done
```

Comment:
Maybe me misreading, but isn't `clusterName` supposed to be searchable field based on above terraform change? Looks more like a nested object here but might be misunderstanding

Reply:
The code is building the JSON object that gets populated into Cosmos, the clusterName in this instance is just a field that is picked up by the dashboard. 
Not sure what you mean by searchable as we wont ever search/query the DB just report the full container contents.

Reply:
Ah - I missed a line which actually builds the object, was just looking at --args and couldn't see clusterName as a field in there